src/main/java/ay2021s1_cs2103_w16_3/finesse/model/ModelManager.java

```diff
@@ -237,12 +241,40 @@ public void updateFilteredTransactionList(Predicate<Transaction> predicate) {
     public void updateFilteredExpenseList(Predicate<Transaction> predicate) {
         requireNonNull(predicate);
         filteredExpenses.setPredicate(t -> predicate.test(t) && PREDICATE_SHOW_ALL_EXPENSES.test(t));
+        refreshTransactionLists();
     }
 
     @Override
     public void updateFilteredIncomeList(Predicate<Transaction> predicate) {
         requireNonNull(predicate);
         filteredIncomes.setPredicate(t -> predicate.test(t) && PREDICATE_SHOW_ALL_INCOMES.test(t));
+        refreshTransactionLists();
+    }
+
+    /**
+     * Casts a FilteredList into a List of the filtered subtype.
+     * Precondition: The given FilteredList must be either the expense list or income list.
+     * This guarantees that the elements of the FilteredList can successfully be cast to the target type.
+     * @param list The FilteredList to be cast.
```

Comment:
```suggestion
     * This guarantees that the elements of the FilteredList can successfully be cast to the target type.
     *
     * @param list The FilteredList to be cast.
```

Add an empty line between the description and parameter section.